### PR TITLE
2021 09 subcell viz go tab

### PR DESCRIPTION
**Pull request by dlrice**
## Purpose
Add GO tab with corresponding sub-cellular location visualisation to entry page [[jira](https://www.ebi.ac.uk/panda/jira/browse/TRM-25568)]

## Approach
- Modify `Tab(s)` component in franklin ([PR](https://github.com/ebi-uniprot/franklin-sites/pull/218)) to allow some tabs to be "cached" - in document but not visible - to prevent having to re-create `@swissprot/swissbiopics-visualizer` and reattach the listeners, tooltips, etc.
- Extract GO data from cross-references.
- Create `uniProtLocationIds` & `goLocationIds` as CSV string in `SubcellularLocationWithVizView` rather than in `SubcellViz` and only pass the appropriate one depending on tab.
- Added `GoEvidenceType` for the future

## To do in future work
- **Location categories**: create and consume sub-cellular location to category mapping [[TRM-26463](https://www.ebi.ac.uk/panda/jira/browse/TRM-26463)]
- **Add source tags**: this will form part of the existing jira [[TRM-23931](https://www.ebi.ac.uk/panda/jira/browse/TRM-23931)]

## Testing
Update snapshots and add unit tests

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

### CODE REVIEW

File: src/uniprotkb/components/protein-data-views/SubCellViz.tsx [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 import { FC, memo, useEffect, useRef } from 'react';
 import tippy from 'tippy.js';
 import '@swissprot/swissbiopics-visualizer';
 
 import 'tippy.js/dist/tippy.css';
 import './styles/sub-cell-viz.scss';
 
-import { SubcellularLocationComment } from '../../types/commentTypes';
+import { RequireExactlyOne } from 'type-fest';
+import { VizTab } from './SubcellularLocationWithVizView';
```
Comment (aurel-l) [inline src/uniprotkb/components/protein-data-views/SubCellViz.tsx:10]: These need to go higher

**Comment by aurel-l** [comment]
Looking good 👍🏽 
Left a comment about the position of an import, nothing more than that.

Comparing with the current website, I'm wondering if we need the `Complete GO annotation on QuickGO ...` link too? It should be fairly easy to add.

File: src/uniprotkb/components/protein-data-views/SubCellViz.tsx (hunk 1 above)
Comment (dlrice) [reply src/uniprotkb/components/protein-data-views/SubCellViz.tsx]: Fixed now

**Comment by dlrice** [comment]
> Comparing with the current website, I'm wondering if we need the `Complete GO annotation on QuickGO ...` link too? It should be fairly easy to add.
Good idea, I didn't even notice it on the current site. Added it in now with an `<hr/>` to separate it from the list of locations.

**Review by aurel-l — APPROVED** [review]
Looking good 👍🏽

**Pull request merged**